docs/CustomCodeTemplates.md

```diff
@@ -0,0 +1,190 @@
+
+Optionally the `react-native-assets` package can be utilized to keep both files in sync.
+
+### Arguments
+Arguments are a structure that represent the configuration of the custom code templates. An `Argument` consists of a `type` and `value`.
+The supported argument types are:
+- Primitives - `boolean`, `number` and `string`. They must have a default value which would be used if no other value is configured for the notification.
+- `object` - An `Object` where keys are the argument names and values are ```Arguments``` with supported primitve values.
+- `file` - a file argument that will be downloaded when the template is triggered. Files don't have default values.
+- `action` - an action argument that could be a function template or a built-in action like ‘close’ or ‘open url’. Actions don't have default values.
+
+#### Hierarchical arguments
+You can group arguments together by either using an `object` argument or indicating the group in the argument's name by using a '.' symbol. Both definitions are treated the same. `file` and `action` type arguments can only be added to a group by specifying it in the name of the argument.
+
+The following code snippets define identical arguments:
+```json
+"arguments": {
+    "map": {
+        "type": "object",
+        "value": {
+            "a": {
+                "type": "number",
+                "value": 5
+            },
+            "b": {
+                "type": "number",
+                "value": 6
+            }
+        }
+    }
+}
+```
+and
+```json
+"arguments": {
+    "map.a": {
+        "type": "number",
+        "value": 5
+    },
+    "map.b": {
+        "type": "number",
+        "value": 6
+    }
+}
+```
+
+### Template definition example
+```json
+"Example template": {
+    "type": "template",
+    "arguments": {
+        "boolArg": {
+            "type": "boolean",
+            "value": false
+        },
+        "stringArg": {
+            "type": "string",
+            "value": "Default"
+        },
+        "mapArg": {
+            "type": "object",
+            "value": {
+                "int": {
+                    "type": "number",
+                    "value": 0
+                },
+                "string": {
+                    "type": "string",
+                    "value": "Default"
+                }
+            }
+        },
+        "actionArg": {
+            "type": "action"
+        },
+        "fileArg": {
+            "type": "file"
+        }
+    }
+}
+```
+
+## Registering custom templates
+
+Templates must be registered within the native applications:
+
+### For Android
+Add the following line in the ```Application.onCreate``` method. If you are extending `CleverTapApplication` add the line before callling ```super.onCreate()```, otherwise add the line before calling ```CleverTapRnAPI.initReactNativeIntegration(this)```
+
+```java
```

Comment:
Can we add this inside the function onCreate() here instead of just the code. Might provide more clarity and easy readbaility?

Reply:
Done